src/gcd_ops.rs

```diff
@@ -0,0 +1,200 @@
+//! Operators for finding greatest common divisor.
+
+use crate::{
+    arch::word::Word,
+    buffer::Buffer,
+    div, gcd,
+    ibig::IBig,
+    memory::MemoryAllocation,
+    ops::{ExtendedGcd, Gcd},
+    ubig::{Repr::*, UBig},
+};
+
+impl Gcd<UBig> for UBig {
```

Comment:
I feel this trait and overloading to take by value are a bit of an overkill, because for large numbers, cloning a number is going to be rather cheap compared to computing the GCD, so there is very small performance benefit to taking by value.

Therefore, let's delete the trait, and make this an inherent function for `UBig`, rather than a trait, and always take both arguments by reference.

Reply:
I think take the argument by value is better? Because in that way the use could choose to take by reference with just a `&` sign.

Reply:
Alright, I guess taking by reference is easier to use for the `self` operand.

Reply:
Yes, taking by reference definitely easier to use. Actually for both operands:
```
a.gcd(&b)
```
easier than:
```
a.clone().gcd(b.clone())
```

Also, if one operand is big and the other tiny, there is no need for any allocations, so pass by reference definitely better.

Reply:
Done in [abb1f1f](https://github.com/tczajka/ibig-rs/pull/11/commits/abb1fbf0764971ed697eceeaface1b12a8d8c003)